### DISCUSSION TITLE
Character animation without rigging

### DISCUSSION THREAD

**Discussion by elkhalafy** [godotengine/godot-proposals#6945] [3D]
### Describe the project you are working on

Hello

How are you? 

I found Idea for character animation without rigging or bones. This idea created by pixar studios called character articulation through profile curves or Curvenent system.

They have been released the paper about this idea, So please if you able to do this As standalone Free application for Windows OS. Or support it inside Godot engine to animate games and films characters. It will be great.

Watch the video and download the paper too

https://youtu.be/gUJ4RxMdVn8

Download the paper here :-

https://graphics.pixar.com/library/ProfileMover/paper.pdf

I hope implement it, Thank you 

### Describe the problem or limitation you are having in your project

Godot engine 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Character animation without rigging pretty needed it for everyone, to make fast animation 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Godot engine 

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Godot engine 

### Is there a reason why this should be core and not an add-on in the asset library?

Inside Godot better than add-on

**Comment by AThousandShips** [comment]
Is this actually feasible as a real time method at the moment? What open source implementations are there of this? Do they target the CPU or the GPU?

I don't think it's realistic to expect an implementation to be developed in core, and as you aren't linking any implementation I'm not sure what you hope/expect with this

This is a highly novel and unproven technology, so without a proper existing *compatible open source* implementation this is not likely to be in core, and even then the lack of support in other software is a major reason not to, even if you were offering to create an implementation

**Comment by elkhalafy** [comment]
In fact the demo reel video explain the idea and the paper too. The paper explains the main concept of idea but unfortunately Pixar didn't release any examples or source code nor library or SDK, just the paper and explains main idea . But the developers can implement it if they accepted and read the paper very well. 

It's not novel nor unproven technology. It's real cause DreamWorks did something like that and all like you Said hidden Bones!!!! And were not hidden Bones. But their technology was private with Intel company. Now Pixar created something Based on curves and you say unproven technology !!! 
But the developer of sculpting inside blender that called Pablo deparro created pose brush to pose any character without depend on bones but not perfect and they made proof of concept animation without rigs but they didn't continue that idea cause lack of features and now Pixar created and animated light-year movie and elemental and next movies based on this idea. cause they rewrite presto software using this unproven technology !!!!

Come on, get real. 

I hope Godot developers implement it in the near future

**Reply by AThousandShips** [reply]
> and all like you Said hidden Bones!!!! And were not hidden Bones

Where did I say that?

> Come on, get real.

Please be civil and professional, and refer to the [code of conduct](https://godotengine.org/code-of-conduct/)

Reacting to someone pointing out issues, and roadblocks to adding it to core, is not an attack on you, or claiming this is any of the things you are saying

The article came out last year, that's what "novel" means, it's new, and have you actually confirmed that the description of the algorithm in the article is actually sufficient to implement this with only minor work to figure out how it works?

**Reply by AThousandShips** [reply]
By "unproven" I mean in the gaming industry, you provided no examples of it being used in game design, as far as I could tell

Godot is *not* a dedicated 3D movie making program, and this kind of animation is meant to be done in some other program and imported most of the time

**Reply by AThousandShips** [reply]
My point, beyond the very obvious issue of there being no open source implementation to build on, or support across other software (at least you haven't presented any), is that for features like this we gain nothing by being ahead of the curve

Godot isn't a modeling program, or an animation program, it's a game engine and editor, supporting things natively that no mainstream modeling software supports just makes it useless

**Reply by Zireael07** [reply]
> supporting things natively that no mainstream modeling software supports just makes it useless

I wouldn't go so far to say it's "useless" but yes it sounds like EXTREMELY significant effort would have to be spent on a feature that would be a better fit in a modeling program.

(Note that Godot can already kind of work as a simple modeling program already with certain add-ons, but complex and specific features are better left to dedicated software)

**Comment by theraot** [comment]
Consider that Godot currently does not have weight painting for rigged meshes, I don't think this belongs in Godot at all. Instead this could be something for Blender. And if it can be baked to blend shape animation, then we can expect Godot to be able to import them and play them.
